providers/azure.go

```diff
@@ -313,6 +361,77 @@ func makeAzureHeader(accessToken string) http.Header {
 	return makeAuthorizationHeader(tokenTypeBearer, accessToken, nil)
 }
 
+func (p *AzureProvider) getGroupsFromProfileAPI(ctx context.Context, s *sessions.SessionState) ([]string, error) {
+	if s.AccessToken == "" {
+		return nil, fmt.Errorf("missing access token")
+	}
+
+	groupsURL := getMicrosoftGraphGroupsURL(p.GroupField).String()
+
+	// Need and extra header while talking with MS Graph. For more context see
+	// https://docs.microsoft.com/en-us/graph/api/group-list-transitivememberof?view=graph-rest-1.0&tabs=http#request-headers
+	extraHeader := makeAzureHeader(s.AccessToken)
+	extraHeader.Add("ConsistencyLevel", "eventual")
+
+	var groups []string
+
+	for groupsURL != "" {
+		// logger.Printf("Calling Group API: %s", groupURL)
+		jsonRequest, err := requests.New(groupsURL).
+			WithContext(ctx).
+			WithHeaders(extraHeader).
+			Do().
+			UnmarshalJSON()
+		if err != nil {
+			return nil, err
+		}
+		groupsURL, err = jsonRequest.Get("@odata.nextLink").String()
+		if err != nil {
+			groupsURL = ""
+		}
+		groupsPage, err := getGroupsFromJSON(jsonRequest, p.GroupField)
+		if err != nil {
+			return nil, err
+		}
+		groups = append(groups, groupsPage...)
+	}
+
+	return groups, nil
+}
+
+func getGroupsFromJSON(json *simplejson.Json, groupField string) ([]string, error) {
+	var groups []string
+	for _, doc := range json.Get("value").MustArray() {
+		for k, v := range doc.(map[string]interface{}) {
```

Comment:
Is doc not a simpleJSON too? I think there are ways to do this with the simpleJSON interface

Reply:
Nope, unfortunately it returns a slice of interfaces, see simplejson's [MustArray](https://godoc.org/github.com/bitly/go-simplejson#Json.MustArray)

Reply:
You could do something like this, use the must array to get an index for iteration but then use the JSON type to actually get the right value
```
for i := range json.Get("value").MustArray() {
  value := json.Get("value").GetIndex(i).Get(groupField).MustString()
  groups = append(groups, value)
}
```

Reply:
Done, thanks.